src/main/java/seedu/address/logic/commands/FindAppCommand.java

```diff
@@ -0,0 +1,59 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+
+import seedu.address.commons.util.ToStringBuilder;
+import seedu.address.logic.Messages;
+import seedu.address.model.Model;
+import seedu.address.model.appointment.AppointmentContainsKeywordsPredicate;
+
+/**
+ * Finds and lists all persons in address book whose name contains any of the argument keywords.
+ * Keyword matching is case insensitive.
+ */
+public class FindAppCommand extends Command {
+
+    public static final String COMMAND_WORD = "findApp";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Finds all appointments whose fields contain any of "
+            + "the specified keywords and displays them as a list with index numbers.\n"
+            + "Parameters: KEYWORD [MORE_OPTIONAL_KEYWORDS]...\n"
+            + "Example: " + COMMAND_WORD + " i/ T0123456A d/ 2024-03-16 from/ 11:00";
```

Comment:
Similar comment to the one above, "contains any of the specified keywords" seem to imply that "from/ 11" will fetch me all appointments from 11am onwards but we do not actually allow for that as 11 is an invalid time. Not quite sure how we can better phrase this as well, maybe we can discuss further?